### PR TITLE
ecompress bugfixes

### PR DESCRIPTION
**Pull request by thesamesam**
…ssion

- Pass -9 like we do for bzip2 & gzip.
- Pass -j from MAKEOPTS for parallel compression.

Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

File: bin/ecompress [hunk 1]
```diff
@@ -128,6 +128,7 @@ if [[ ${PORTAGE_COMPRESS_FLAGS+set} != "set" ]] ; then
 		# See: https://bugs.gentoo.org/672916
 		# Setting '--rm' will remove the source files after a successful compression.
 		lz4)  PORTAGE_COMPRESS_FLAGS="-m --rm";;
+		xz)   PORTAGE_COMPRESS_FLAGS="-9 -T$(__makeopts_jobs)";;
```
Comment (mgorny) [inline bin/ecompress:131]: I have somewhat mixed feelings about this. On one hand, we're passing `-9` to maximize compression. On the other, we're now defaulting to an arbitrary number of jobs that actually worsens compression. But I guess most people won't really care, and it has probably better gain-to-cost ratio than going for lower compression level.

File: bin/ecompress [hunk 2]
```diff
@@ -71,9 +71,9 @@ while [[ $# -gt 0 ]] ; do
 				# note: to save time, we need to do this only if there's
 				# at least one compressed file
 				case ${path} in
-					*.Z|*.gz|*.bz2|*.lzma|*.xz)
+					*.Z|*.gz|*.bz2|*.lzma|*.xz|*.zst)
```
Comment (mgorny) [inline bin/ecompress]: Don't we want `.lz4`, `.lz`, `lzo` here as well?

File: bin/ecompress (hunk 1 above)
Comment (thesamesam) [reply bin/ecompress:131]: Right, people can still override it if they want.

File: bin/ecompress (hunk 2 above)
Comment (thesamesam) [reply bin/ecompress]: Good point, I'll add that now.

**Comment by thesamesam** [comment]
Merged.

**Pull request closed without merging**